### ISSUE TITLE
action :remount in mount resource is not idempotent. 

### ISSUE THREAD

**Issue by kbaghele** [chef/chef#13494]
## Description
For action `:remount`, `mount` resource  is not idempotent by default.

Note: 
This is the way it should work:
If remount then it should check if mounted or not. 
If not then it need to remount else do nothing but a log message.

## Chef Version
Chef Infra Client: 17.10.3

## Platform Version
Alma Linux 8

## Replication Case
Example
my code:
```ruby
mount tmp do
    fstype 'tmpfs'
    device tmp
    options 'nosuid,nodev,noexec'
    action :remount
    not_if 'mount | grep /tmp | grep nosuid,nodev,noexec'
 end
```

Cookbook Result on 1st run:

```
mount[/tmp] action remount
unmount /tmp
mount /tmp
```

Cookbook Result on 2nd run:

```
mount[/tmp] action remount
unmount /tmp
mount /tmp
```

Note:  Currently to male it idempotent I am adding not_if guard.

## Client Output
```
mount[/tmp] action remount
unmount /tmp
mount /tmp
```

**Comment by neha-p6** [comment]
The recipe used here is incorrect. 

The `mount` resource has a property named supports which sets {remount: false} by default. Ref  doc [mount Resource](https://docs.chef.io/resources/mount/#properties) , and by definition of this resource if this flag is not set for action remount , it does the unmounting and then mounting of the endpoint.
Here’s action remount https://github.com/chef/chef/blob/main/lib/chef/provider/mount.rb#L62, if it does not find property `supports` to be set to true, it will perform the recipe by first unmounting and then mounting the endpoint(which is what is seen example mentioned above). So recipe rather needs to set this property for the mount resource to be able to do a remount Once the property is set, the resource will continue with the remount_fs call [here](https://github.com/chef/chef/blob/main/lib/chef/provider/mount/mount.rb#L140) and perform the mount -o remount... command which is not happening due to the missing property.

So in conclusion, it is not an issue in the mount resource but the recipe is incorrect. The expected recipe for action remount to perform remounting of an existing endpoint is to set the property `supports remount: true` in the recipe.
